app/models/nic/base.rb

```diff
@@ -33,6 +33,8 @@ class Base < ActiveRecord::Base
     validates :domain_id, :presence => true, :if => Proc.new { |nic| nic.host && nic.host.managed? && nic.primary? }
     validates :ip, :presence => true, :if => Proc.new { |nic| nic.host && nic.host.managed? && nic.require_ip_validation? }
 
+    validate :validate_host_taxonomy
```

Comment:
The validation code itself is guarded:
```ruby
      return true if self.subnet.nil?
      return true if self.host.organization.nil? && self.subnet.organizations.empty? &&
                     self.host.location.nil? && self.subnet.locations.empty?
```
Besides, it's a pretty big issue that is true for the whole system - if the taxonomy is tuned off, you can do some changes that will break consistency of the DB, if you'll turn the taxonomy back on.